### PR TITLE
toolkit(assistants): dark mode buttons

### PR DESCRIPTION
**Pull request by abimacarmes**
Thank you for contributing to the Cohere Toolkit!

Cell Buttons:

https://github.com/user-attachments/assets/1b90d0ab-d897-40c2-a4d9-f521c2e837ac

Primary:

https://github.com/user-attachments/assets/5af265f3-5739-40c4-9b7c-a3169928eb29

Outline:

https://github.com/user-attachments/assets/6a83b0e8-2f78-459d-a510-fadaf8bf05da

Secondary:

https://github.com/user-attachments/assets/179b3022-59ea-48c0-b546-8edd451a2794

- [ ] **PR title**: "area: description"
  - Where "area" is whichever of interface, frontend, model, tools, backend, etc. is being modified. Use "docs: ..." for purely docs changes, "infra: ..." for CI changes.
  - Example: "deployment: add Azure model option"

- [ ] **PR message**: ***Delete this entire checklist*** and replace with
    - **Description:** a description of the change
    - **Issue:** the issue # it fixes, if applicable
    - **Dependencies:** any dependencies required for this change

- [ ] **Add tests and docs**: Please include testing and documentation for your changes
- [ ] **Lint and test**: Run `make lint` and `make run-tests`

**AI Description**

This PR updates the button components in the assistants_web interface. The changes include:

- Replacing `type` with `buttonType` and adding `kind` and `iconPosition` attributes to the `Button` component in the `Login` and `Register` forms.
- Replacing `endIcon` with `icon` and `iconPosition` in multiple components, including `AgentToolFilePicker`, `DiscoverAgentCard`, `DeleteAgent`, `ComposerError`, `MessagingContainer`, `ReadOnlyConversation`, and `Settings/ToolsInfoBox`.
- Replacing `kind="secondary"` with `kind="cell"` in the `Button` component of `MessagingContainer.
- Replacing `kind="green"` with `kind="secondary"` and `theme="evolved-green"` in the `Button` component of `CreateAgent`.
- Replacing `kind="danger"` with `label` and `iconPosition="end"` in the `Button` component of `DeleteAgent`.
- Replacing `kind="minimal"` with `kind="secondary"` in the `Button` component of `EditEnvVariablesButton`.
- Replacing `kind="secondary"` with `kind="cell"` in the `Button` component of `DeleteConversations`.
- Replacing `kind` and `endIcon` with `theme="mushroom-marble"` and `cellButton` in the `AuthLink` component of `NavigationUserMenu`.
- Replacing `kind="primary"` with `kind="cell"` and adding `icon` in the `Button` component of `NavigationUserMenu`.
- Replacing `kind="secondary"` with `kind="cell"` and adding `icon` in the `Button` component of `Settings/AgentsToolsTab`.
- Replacing `kind="secondary"` with `kind="cell"` and adding `icon` and `isLoading` in the `Button` component of `Modals/EditConversationTitle`.
- Replacing `kind="secondary"` with `kind="cell"` and adding `icon` in the `Button` component of `ShareModal`.
- Replacing `kind="secondary"` with `kind="cell"` and adding `icon` in the `Button` component of `WebSearchConfigurationModal`.
- Replacing `kind="secondary"` with `kind="cell"` and `iconPosition="end"` in the `Button` component of `ConnectDataModal.
- Replacing `kind="secondary"` with `kind="outline"` and `theme="mushroom-marble"` in the `Button` component of `ButtonGroup`.
- Replacing `kind="secondary"` with `danger` and `label` in the `Button` component of `AuthLink`.
- Replacing `kind="primary"` with `kind="cell"` and adding `icon` in the `Button` component of `AuthLink`.
- Replacing `kind="secondary"` with `kind="cell"` and adding `icon` in the `Button` component of `AgentsSidePanel`.

### CODE REVIEW

File: src/interfaces/assistants_web/src/components/Shared/Button/Button.tsx [hunk 1]
```diff
@@ -1,128 +1,288 @@
...
+  disabled?: boolean;
+  isLoading?: boolean;
+  className?: string;
+  iconPosition?: 'start' | 'end';
+  iconOptions?: {
+    className?: string;
+    kind?: 'default' | 'outline';
+    customIcon?: React.ReactNode;
+  };
+  buttonType?: 'submit' | 'reset' | 'button' | undefined;
```
Comment (knajjars) [inline src/interfaces/assistants_web/src/components/Shared/Button/Button.tsx:90]:
```suggestion
  buttonType?: 'submit' | 'reset' | 'button';
```

File: src/interfaces/assistants_web/src/components/Shared/Button/Button.tsx [hunk 2]
```diff
@@ -1,128 +1,288 @@
...
+  onClick?: React.MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>;
+  href?: string;
+  hrefOptions?: {
+    shallow?: boolean;
+    rel?: string;
+    target?: HTMLAttributeAnchorTarget;
+  };
+  animate?: boolean;
```
Comment (knajjars) [inline src/interfaces/assistants_web/src/components/Shared/Button/Button.tsx:98]: do we still need this? I think we want everything to be animated

File: src/interfaces/assistants_web/src/components/Shared/Button/Button.tsx [hunk 3]
```diff
@@ -1,128 +1,288 @@
...
+  iconOptions?: {
+    className?: string;
+    kind?: 'default' | 'outline';
+    customIcon?: React.ReactNode;
+  };
+  buttonType?: 'submit' | 'reset' | 'button' | undefined;
+  onClick?: React.MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>;
+  href?: string;
+  hrefOptions?: {
+    shallow?: boolean;
```
Comment (knajjars) [inline src/interfaces/assistants_web/src/components/Shared/Button/Button.tsx:94]: app router does not have shallow links, we should remove this

File: src/interfaces/assistants_web/src/components/Shared/Button/Button.tsx [hunk 4]
```diff
@@ -1,128 +1,288 @@
...
+  disabled = false,
+  isLoading = false,
+  className,
+  iconOptions,
+  buttonType,
+  onClick,
+  href,
+  hrefOptions,
+  animate = true,
+  danger = false,
```
Comment (knajjars) [inline src/interfaces/assistants_web/src/components/Shared/Button/Button.tsx:117]: do we need this? Can `danger` be a `kind` instead or a theme

File: src/interfaces/assistants_web/src/components/Shared/Button/Button.tsx (hunk 2 above)
Comment (abimacarmes) [reply src/interfaces/assistants_web/src/components/Shared/Button/Button.tsx:98]: I think it's good to leave it in - for icon buttons we don't usually want them to animate.

**Review by knajjars — APPROVED** [review]

**Pull request merged**